### DISCUSSION TITLE
Will RefreshMemoryLimit ship as preview

### DISCUSSION THREAD

**Discussion by normj** [dotnet/runtime#91567] [General] [answered]
I'm looking into integrating the new `RefreshMemoryLimit` API into the [AWS Lambda .NET runtime](https://github.com/dotnet/runtime/issues/70601#issuecomment-1555383466). I see the API is marked as preview. Will the API ship as part of .NET 8 as preview? If so I'm uncomfortable updating our production Lambda bootstrapping code to use a preview API. I was really hoping to use this new API to address the issue .NET does not recognize the correct amount of memory inside AWS Lambda.

@mangod9 @cshung

**Comment by cshung** [comment]
It would be great if you could confirm this worked for you, with that, I am more than happy to take the preview label off!

**Comment by normj** [comment]
I can confirm using preview 7 the APIs work for AWS Lambda's use case. I have a PR out in our repository using the new API and the test pass for .NET 8 and fail for .NET 6. https://github.com/aws/aws-lambda-dotnet/pull/1578

**Comment by cshung** [comment, marked as answer]
This is the PR to take the preview label off - https://github.com/dotnet/runtime/pull/91622

**Comment by normj** [comment]
Great, thanks! I'll follow the PR. Would the PR make RC1 or is it too late for that?

**Reply by teo-tsirpanis** [reply]
I don't think so; the last commit to RC1 was two weeks ago.

**Reply by normj** [reply]
Thanks, I figured it was too late for RC1 but figured should double check.

**Comment by normj** [comment]
@cshung I just test this feature in .NET 8 RC2 and was able to remove my pragma to disable the API preview errors. Thanks!

Also I noticed in the .NET 8 what's new post it talks about this API but it is out of date saying the API is private and requires reflection to call it. I submitted feedback that it was out of date but thought I would let you know as well. https://learn.microsoft.com/en-us/dotnet/core/whats-new/dotnet-8#garbage-collection

**Reply by cshung** [reply]
Thanks for the notice!

The post should now be updated with https://github.com/dotnet/docs/pull/37474 by @MokoSan, thanks!

If there is a feedback item for that, it should be closed with that PR as well.
